azure-pipelines.yml

```diff
@@ -1,19 +1,42 @@
-# Starter pipeline
-# Start with a minimal pipeline that you can customize to build and deploy your code.
-# Add steps that build, run tests, deploy, and more:
-# https://aka.ms/yaml
-
 trigger:
 - main
+- master
+- staging/*
+- dev/*
+
+pr:
+- main
+- master
 
-pool:
-  vmImage: 'ubuntu-latest'
+variables:
+  isMain: $[eq(variables['Build.SourceBranch'], 'refs/heads/master')]
+  runCondition: true
+  targetBranch: $[ variables['Build.SourceBranchName'] ]
 
-steps:
-- script: echo Hello, world!
-  displayName: 'Run a one-line script'
+jobs:
+- job: LinuxBuilds
+  strategy:
+    matrix:
+      astyle:
+        imageName: 'ubuntu-latest'
+        BUILD_TYPE: astyle
+        TARGET_BRANCH: '$(targetBranch)'
+      cppcheck:
+        imageName: 'ubuntu-latest'
+        BUILD_TYPE: cppcheck
+        TARGET_BRANCH: '$(targetBranch)'
+      tinyiiod:
+        imageName: 'ubuntu-latest'
+        BUILD_TYPE: tinyiiod
 
-- script: |
-    echo Add other tasks to build, test, and deploy your project.
-    echo See https://aka.ms/yaml
-  displayName: 'Run a multi-line script'
+  pool:
+      vmImage: $(imageName)
+  steps:
+  - checkout: self
+    condition: eq(variables.runCondition, true)
+    fetchDepth: 50
+    clean: true
+    persistCredentials: true
```

Comment:
is this pushing back anything?
if not, i would not leave the credentials stored;